### PR TITLE
Support for spider_syn

### LINKED ISSUE #58: Add Spider-Syn dataset
See https://github.com/ygan/Spider-Syn

### PR DESCRIPTION
**Pull request by salokr**
Hi @tscholak,

I have added support for Spider-Syn ( I see that somebody has already finished for real ). I was waiting for the [bug](https://github.com/ElementAI/picard/pull/github.com/ElementAI/picard/issues/61) to be resolved but now I guess we can resolve this by bypassing the local reference to URL_ variable for the datasets as suggested here

Here are my number on all versions of Spider:
Screen Shot 2022-03-04 at 2 09 06 PM

**Note that the questions are specified in the "SpiderSynQuestion" field of the json.**

closes #58

### CODE REVIEW

**Comment by tscholak** [comment]
Thank you @salokr!
Thanks also to @kotalarag

It turns out we can merge spider, spider-realistic, spider-dyn, etc. into one dataset class by following the approach of this dataset: https://github.com/cccntu/datasets/blob/fdc388e782923083060bf4ed5d2b0d83ada41189/datasets/xor_tydi_qa/xor_tydi_qa.py. I will look into that once everything is merged.

**Review by tscholak — APPROVED** [review]
Thanks so much! LGTM!

**Comment by tscholak** [comment]
@salokr can you remove the `.DS_Store` files, please?

**Comment by salokr** [comment]
I deleted the .DS_Store file.
Thanks

**Comment by tscholak** [comment]
@salokr there's another one 😝 
can I ask you to remove all of them?

**Comment by salokr** [comment]
There were two it seems :) Deleted them

File: .gitignore [hunk 1]
```diff
@@ -148,3 +148,4 @@ eval/*
 transformers_cache/*
 wandb/*
 database/*
+.DS_Store
```
Comment (tscholak) [inline .gitignore:151]: thanks!

**Comment by tscholak** [comment]
It's merged, all good!
Thanks!

**Pull request merged**